lib/Basic/FileInfo.cpp

```diff
@@ -14,22 +14,93 @@
 
 #include <cassert>
 #include <cstring>
+
+#if defined(_WIN32)
+#include <windows.h>
+#include <direct.h>
+#else
 #include <sys/stat.h>
+#endif
 
 using namespace llbuild;
 using namespace llbuild::basic;
 
 bool FileInfo::isDirectory() const {
-  return mode & S_IFDIR;
+  return (mode & S_IFDIR) != 0;
 }
 
+#if defined(_WIN32)
+static void FileTimeToTimestamp(FILETIME fileTime, FileTimestamp *stamp) {
```

Comment:
The reason why I decided to add this conversion helper is because `FileTimestamp` is defined as follows
```
struct FileTimestamp {
  uint64_t seconds;
  uint64_t nanoseconds;
}
```

> In that case, we could simply decide to document that and not bother to put that conversion here. What do you think?

The concern I have with this is that we'd have to change `FileTimestamp` into some kind of platform specific entity. This would require explicitly defining all the comparison operators again, and would be complicated. 

I'm not really sure if its overkill or not for this. That said, you could probably call the current method overkill. Up to you!